code/gateway/dataaccessOverwrite.q

```diff
@@ -0,0 +1,71 @@
+\d .dataaccess
+
+// Dynamic routing finds all processes with relevant data
+attributesrouting:{[options;procdict]
+    // Get the tablename and timespan
+    timespan:`long$options[`starttime`endtime];
+    //adjust rdb partition range to account for period end time)
+    if[`rdb in key procdict;procdict[`rdb]:(first[procdict `rdb];-1+`long$01D00 + last procdict `rdb)];
+    // See if any of the provided partitions are with the requested ones
+    procdict:{[x;timespan] (all x within timespan) or any timespan within x}[;timespan] each procdict;
+    // Only return appropriate dates
+    types:(key procdict) where value procdict;
+    // If the dates are out of scope of processes then error
+    if[0=count types;
+        '`$"gateway error - no info found for that table name and time range. Either table does not exist; attributes are incorect in .gw.servers on gateway, or the date range is outside the ones present"
+       ];
+    :types;
+    };
+
+// function to adjust the queries being sent to processes to prevent overlap of
+// time clause and data being queried on more than one process
+adjustqueries:{[options;part]
+    // if only one process then no need to adjust
+    if[2>count p:options`procs;:options];
+    // get the partitions that are required by each process
+    tabname:options[`tablename];
+    // Remove duplicate servertypes from the gw.servers
+    servers:select from .gw.servers where i=(first;i)fby servertype;
+    // extract the procs which have the table defined
+    servers:select from servers where {[x;tabname]tabname in @[x;`tables]}[;tabname] each attributes;
+    // Create a dictionary of the attributes against servertypes
+    procdict:(exec servertype from servers)!(exec attributes from servers)@'(key each exec attributes from servers)[;0];
+    // If the response is a dictionary index into the tablename
+    procdict:@[procdict;key procdict;{[x;tabname]if[99h=type x;:x[tabname]];:x}[;tabname]];
+    //create list of all possible partitions
```

Comment:
minor niggle, but inconsistency in the comments - some have space at start, some don't. some start with upper case, some with lower case - my preference is a space & lower case, but just make them all consistent please